R/report-plots.R

```diff
@@ -300,3 +300,390 @@ district_barplot <- function(naomi_output,
+      updatemenus = list(
+        list(
+          type = 'dropdown',
+          y = 1.2,
+          x  = 0.2,
+          active = 0,
+          buttons = buttons
+        )
+      )
+    ) %>%
+    config(modeBarButtonsToRemove = remove_buttons, displaylogo = FALSE)
+
+  suppressWarnings(final_plot)
+
+}
+
+#' Plotly scatterplot data inputs and naomi outputs
+#'
+#' @param df Inputs_outputs dataframe containing matched model estimates and data inputs.
+#' @param ind Indicator filter.
+#' @param quarter Calendar quarter filter.
+#' @param age_disag Age group filter.
+#' @param sex_disag Sex filter.
+#' @param input_data Input data name.
+#' @param input_data_type Input data type.
+#'
+#' @export
+
+scatter_plotly <- function(df,
+                           ind,
+                           quarter,
+                           input_data,
+                           input_data_type,
+                           age_disag = "Y015_049",
+                           sex_disag  = "both") {
+
+  suppressMessages(library(plotly))
+
+  remove_buttons <- c("zoomIn2d", "zoomOut2d", "pan2d", "select2d", "lasso2d",
+                      "autoScale2d", "resetScale2d", "hoverClosestCartesian",
+                      "hoverCompareCartesian", "zoom")
+
+
+  input_source <- paste0(input_data, " ", quarter)
+  output_source <- paste0("Naomi estimate ", quarter)
+
+
+  plot_data <- df %>%
+    dplyr::filter(indicator == ind,
+                  calendar_quarter == quarter,
+                  age_group == age_disag,
+                  sex == sex_disag,
+                  !is.na(mean))
+
+
+  # Get ranges for axis
+  if(grepl("prevalence", ind)){
+    max <- max(plot_data$mean) + 0.02
+    range <- "5"
+    title <- "HIV prevalence"
+  }
+
+  if(grepl("art_coverage", ind)){
+    max <- max(plot_data$mean) + 0.05
+    min <- min(plot_data$mean) - 0.05
+    range <- "10"
+    title <- "ART coverage"
+  }
+
+  # Filter for correct survey in case of multiple surveys
+  if(input_data_type == "survey"){
+    plot_data <- plot_data %>%
+      dplyr::filter(source %in% c(input_source, output_source))
+  }
+
+
+  age_groups <- get_age_groups()
+  age_label <- age_groups[age_groups$age_group == unique(plot_data$age_group), ]$age_group_label
+  if(sex_disag == "both"){sex_label = "all"}else{sex_label = sex_disag}
+
+  plot_title <- paste("<b>", title, ": ", input_data, " vs. ", output_source,
+                      "</br><sub> (", sex_label, ",", age_label,")</sub><br>")
+
+  if(grepl("anc", ind)) {output_source <- paste0(output_source, " females 15-49")}
+
+
+  plot_data_wide <- plot_data %>%
+    dplyr::mutate(data_type = ifelse(grepl("Naomi estimate",source), "output", "input")) %>%
+    dplyr::select(area_id, area_name, area_level_label, sex, age_group,
+                  calendar_quarter, indicator, mean, data_type) %>%
+    tidyr::pivot_wider(names_from = data_type, values_from = mean) %>%
+    filter(!is.na(input))
```

Comment:
This line is failing from hintr but I've not been able to have a look into why it is failing yet